Sources/web3swift/Contract/EthereumContract.swift

```diff
@@ -6,137 +6,140 @@
 import Foundation
 import BigInt
 
-public struct EthereumContract: ContractProtocol {
+public class EthereumContract: ContractProtocol {
```

Comment:
Do we want to keep to user ability to inherits this type or should we consider to make it `final`?

Reply:
I think we do want to keep it open. It looks like `EthereumContract` will serve as a good base for future implementations of some ERCs as ABI decoding is happening in one place only.